tools/src/argparseUtils.h

```diff
@@ -0,0 +1,42 @@
+#ifndef PODIO_TOOLS_ARGPARSEUTILS_H // NOLINT(llvm-header-guard): folder structure not suitable
+#define PODIO_TOOLS_ARGPARSEUTILS_H // NOLINT(llvm-header-guard): folder structure not suitable
+
+#include <algorithm>
+#include <charconv>
+#include <iostream>
+#include <ranges>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <vector>
+
+template <typename... Flags>
+auto findFlags(const std::vector<std::string>& argv, Flags... flags) {
+  return std::find_if(argv.begin(), argv.end(), [&](const auto& elem) { return ((elem == flags) || ...); });
+}
+
+inline std::vector<std::string_view> splitString(const std::string& str, const char delimiter) {
+  std::vector<std::string_view> tokens;
+
+  for (const auto& token : std::ranges::views::split(str, delimiter)) {
+    tokens.emplace_back(token.begin(), token.end());
+  }
+
+  return tokens;
+}
+
+inline size_t parseSizeOrExit(std::string_view str) {
+  try {
+    std::size_t number{};
+    const auto [ptr, err] = std::from_chars(str.data(), str.data() + str.size(), number);
+    if (err != std::errc{} || ptr != str.data() + str.size()) {
+      throw std::invalid_argument("");
+    }
+    return number;
+  } catch (const std::invalid_argument&) {
+    std::cerr << "'" << str << "' cannot be parsed into an integer number" << std::endl;
```

Comment:
I've noticed that now passing a bad range like `-e 1:` fails with `'' cannot be parsed into an integer number` while before this would fail with: 
```
usage: podio-dump [-h] [-c CATEGORY] [-e ENTRIES] [-d] [--dump-edm DUMP_EDM] [--version]
                  inputfile
podio-dump: error: argument -e/--entries: '0:' cannot be parsed into a list of entries
```
Not a huge problem since it is stated in the help how to correctly pass a range, but the error message is a bit useless in that case.

Reply:
Should be back to (almost) what it was before now. I have put in an explicit check for an empty string after the colon. Ideally `parseSizeOrExit` would return an optional, but that would require quite some other work as well.